python/GafferTest/FileSystemPathTest.py

```diff
@@ -304,6 +304,40 @@ def testMixedSlashes( self ) :
 			self.assertEqual( list( p ), [ r"unc\tshare", r"must\twork" ] )
 			self.assertEqual( p.nativeString(), r"/unc\tshare/must\twork")
 
+	def testEscapedCharacters( self ) :
+
+		p = Gaffer.FileSystemPath( r"/\~this/\#must/\$be/escaped.txt" )
+		if os.name == "nt" :
+			self.assertEqual( list( p ), [ r"\#must", r"\$be", "escaped.txt" ] )
+			self.assertEqual( p.root(), r"//\~this/")
+			self.assertEqual( str( p ), r"//\~this/\#must/\$be/escaped.txt" )
+			self.assertEqual( p.nativeString(), r"\\\~this\\#must\\$be\escaped.txt" )
+		else :
+			self.assertEqual( list( p ), [ "\\~this", "\\#must", "\\$be", "escaped.txt" ] )
+			self.assertEqual( p.root(), "/")
+			self.assertEqual( str( p ), "/\\~this/\\#must/\\$be/escaped.txt" )
+			self.assertEqual( p.nativeString(), "/\\~this/\\#must/\\$be/escaped.txt" )
```

Comment:
It's a tiny bit mindbending that we're not using raw strings (`r""`) for these assertions as well. It makes them harder to verify against the raw string from the input.